pyro/infer/mcmc/nuts.py

```diff
@@ -102,7 +103,8 @@ def _build_basetree(self, z, r, z_grads, log_slice, direction, energy_current):
         tree_size = 1 if sliced_energy <= 0 else 0
         diverging = sliced_energy >= self._max_sliced_energy
```

Comment:
I think that we should change this to `diverging = not (sliced_energy < self._max_sliced_energy)`